### PR TITLE
Adding Spanner STRUCT param samples

### PR DESCRIPTION
**Pull request by jabubake**

### CODE REVIEW

**Comment by jabubake** [comment]
@jsimonweb FYI

File: spanner/cloud-client/snippets.py [hunk 1]
```diff
@@ -598,6 +599,164 @@ def query_data_with_timestamp(instance_id, database_id):
...
+
+    with database.snapshot() as snapshot:
+        results = snapshot.execute_sql(
+            "SELECT SingerId FROM Singers WHERE "
+            "(FirstName, LastName) = @name",
+            params={'name': record_value},
+            param_types={'name': record_type})
+
+    for row in results:
+            print(u'SingerId: {}'.format(*row))
```
Comment (theacodes) [inline spanner/cloud-client/snippets.py]: nit: this is indented too far.

File: spanner/cloud-client/snippets.py [hunk 2]
```diff
@@ -598,6 +599,164 @@ def query_data_with_timestamp(instance_id, database_id):
...
+
+def query_with_array_of_struct(instance_id, database_id):
+    """
+       Query a table using an array of STRUCT parameters.
+    """
+    # [START spanner_create_user_defined_struct]
+    name_type = param_types.Struct([
+        param_types.StructField('FirstName', param_types.STRING),
+        param_types.StructField('LastName', param_types.STRING)
+        ])
```
Comment (theacodes) [inline spanner/cloud-client/snippets.py]: nit: this is indented one too far.

File: spanner/cloud-client/snippets.py [hunk 3]
```diff
@@ -598,6 +599,164 @@ def query_data_with_timestamp(instance_id, database_id):
 # [END spanner_query_data_with_timestamp_column]
 
 
+# [START spanner_write_data_for_struct_queries]
+def write_struct_data(instance_id, database_id):
+    """Inserts sample data that can be used to test STRUCT parameters in queries.
```
Comment (kurtisvg) [inline spanner/cloud-client/snippets.py]: Minor nit: docstrings are inconsistent in terms of spacing/line breaks between functions.

Comment (jabubake) [reply spanner/cloud-client/snippets.py]: Ack, done.

File: spanner/cloud-client/snippets.py (hunk 1 above)
Comment (jabubake) [reply spanner/cloud-client/snippets.py]: done.

File: spanner/cloud-client/snippets.py (hunk 2 above)
Comment (jabubake) [reply spanner/cloud-client/snippets.py]: done.

**Pull request merged**